### ISSUE TITLE
Text should resize itself after reaching MaxLines not after 1 line

### ISSUE THREAD

**Issue by Bawen34** [lzhuor/auto_size_text_field#35]
Hello,
When using the widget in multiline with a maxLines parameter, the text resize itself after reaching the end of the first line, not after reaching the maxLines
```
AutoSizeTextField(
              controller: sessionDescription,
              maxLines: 3,
              minLines: 3,
              keyboardType: TextInputType.multiline,
              decoration: InputDecoration(                
                  alignLabelWithHint: true,               
                  labelText: 'Description'),             
              textAlign: TextAlign.left,
              minFontSize: 12,
            ),
```

**Comment by TDSSSzero** [comment]
yeah,I have this question too. This behavior is not consistent with the auto_size_text

**Comment by anders-au** [comment]
This still remains an issue. Has anyone found a good workaround or fix?

**Comment by tristecsv** [comment]
Hi! 👋

I was able to reproduce the issue.

When using `maxLines > 1`, the font size starts decreasing as soon as the text overflows the first line, instead of waiting until all available lines are used.

### Expected behavior
The text should wrap normally up to `maxLines`, and only then start reducing the font size if it still overflows.

### Possible cause
This seems related to `_checkTextFits`, which likely treats first-line overflow as a failure without considering remaining lines.

### Workaround 
At the moment, there isn’t a clean workaround other than avoiding multiline auto-sizing or manually adjusting layout constraints.

---

I can work on a fix if the intended behavior is to match `auto_size_text`.

@lzhuor could you confirm if this behavior is intended, or should it match `auto_size_text`?

Thanks! 👍
